app/src/main/java/com/greenfox/peridot/peridot_coz_android/activity/MainActivity.java

```diff
@@ -56,14 +81,88 @@ public void checkSharedPreferencesForUser(User user) {
         }
     }
 
-    public void logout(View v){
-        SharedPreferences preferences =getSharedPreferences("userInfo", Context.MODE_PRIVATE);
+    public void logout(){
+        SharedPreferences preferences = this.getSharedPreferences("userInfo", Context.MODE_PRIVATE);
         SharedPreferences.Editor editor = preferences.edit();
         editor.putString("username", "");
         editor.putString("password", "");
         editor.apply();
         Toast.makeText(this,"Successful logout", Toast.LENGTH_SHORT).show();
-        Intent loginIntent = new Intent(MainActivity.this, LoginActivity.class);
-        startActivity(loginIntent);
+
+        Intent loginIntent = new Intent(this, LoginActivity.class);
+        this.startActivity(loginIntent);
```

Comment:
@radaladar `this` is not needed again, and also you can just instantiate the intent as the parameter